packages/adobe-sign/src/apis/adobe-sign-api.ts

```diff
@@ -1,19 +1,58 @@
-import { AwsSecretKey, AwsSecretName, SecretsManagerService } from "asu-core";
+import {AwsSecretKey, AwsSecretName, AxiosProvider, SecretsManagerService} from "asu-core";
 import { injectable } from "tsyringe";
 import { Agreement } from "../models/agreement";
+import {Axios, AxiosError} from "axios";
+import {ApiError} from "./errors/api-error";
+import {AdobeApiError} from "./errors/adobe-api-error";
+import {InternalApiError} from "./errors/internal-api-error";
 
 @injectable()
 export class AdobeSignApi {
-    constructor(private secretsManagerService: SecretsManagerService) {}
+    private httpClient: Axios;
+    constructor(private secretsManagerService: SecretsManagerService, private axiosProvider: AxiosProvider) {}
+
+    private async getHttpClient(): Promise<Axios> {
+        if (!this.httpClient) {
+            try {
+                console.log('Fetching integration key from secrets manager');
+                const integrationKey = await this.secretsManagerService.getSecret(
+                    AwsSecretName.AdobeSign,
+                    AwsSecretKey.AdobeSignIntegrationKey
+                );
+
+                console.log('Creating axios instance');
+                this.httpClient = this.axiosProvider.resolve({
+                    //@todo get Base URL from cache/secretsManager
+                    baseURL: 'https://api.na3.adobesign.com/api/rest/v6',
+                    headers: {
+                        Authorization : `Bearer ${integrationKey}`
+                    }
+                });
+            } catch ( err ) {
+                throw new InternalApiError('Could not initialize AdobeSign HTTP client. ' + err.message);
+            }
+        }
+
+        return this.httpClient;
+    }
 
     public async getAgreement(id: string): Promise<Agreement> {
-        const integrationKey = await this.secretsManagerService.getSecret(
-            AwsSecretName.AdobeSign,
-            AwsSecretKey.AdobeSignIntegrationKey
-        );
-        const agreement = new Agreement();
-        agreement.id = id;
+        try {
+            const httpClient = await this.getHttpClient();
+            console.log('Fetching agreement from AdobeSign');
+            const agreementRes = await httpClient.get(`/agreements/${id}`);
+            return new Agreement(agreementRes.data);
+        } catch ( err ) {
+            AdobeSignApi.handleAdobeApiErrors(err);
+        }
+    }
+
+    private static handleAdobeApiErrors(error: ApiError|AxiosError) {
+        if (error instanceof ApiError) {
+            // If already an ApiError (e.g. InternalApiError), forward error
+            throw error;
+        }
 
-        return agreement;
+        throw new AdobeApiError(error.response.status, error.response.data.message)
```

Comment:
I am not sure how I feel about returning the Adobe status code back to our requester. If we fail to auth with Axios, we should return a 500. If we return a 401, the requester will think their credentials are wrong, but really its a config issue with our app.